gather_metrics.py

```diff
@@ -0,0 +1,114 @@
+import socket
+import sys
+import getopt
+from datetime import datetime
+import subprocess
+
+import conf_collectd_plugin as conf
+import install_utils as utils
+import config
+
+# Python required base version, haven't tested 3 yet.
+REQ_VERSION = (2, 7)
+
+
+def check_version():
+    cur_version = sys.version_info
+    utils.print_step("  Checking python version")
+    if cur_version < REQ_VERSION:
+        sys.stderr.write(
+            "Your Python interpreter is older " +
+            "than what we have tested, we suggest upgrading " +
+            "to a newer 2.7 version before continuing.\n")
```

Comment:
You can use a multiline string here, or, did you know that python automatically concatenates adjacent strings within parens?  Meaning the `+`s are actually unnecessary